arrow-core-test/src/main/kotlin/arrow/core/test/laws/FxLaws.kt

```diff
@@ -0,0 +1,166 @@
+private typealias SuspendFxBlock<F, A> = suspend (suspend BindSyntax<F>.() -> A) -> Kind<F, A>
+
+object FxLaws {
+
+  fun <F, A> laws(
+    pureGen: Gen<Kind<F, A>>, // TODO cannot specify or filter a pure generator, so we need to require an additional one
+    G: Gen<Kind<F, A>>,
+    fxBlock: EagerFxBlock<F, A>,
+    sfxBlock: SuspendFxBlock<F, A>
+  ): List<Law> = listOf(
+    Law("non-suspended fx is lazy") { nonSuspendedIsLazy(G, fxBlock) },
+    Law("suspended fx is lazy") { suspendedIsLazy(G, sfxBlock) },
+    Law("non-suspended fx can bind immediate values") { nonSuspendedCanBindImmediate(G, fxBlock) },
+    Law("non-suspended fx can bind immediate exceptions") { nonSuspendedCanBindImmediateException(pureGen, fxBlock) },
+    Law("suspended fx can bind immediate values") { suspendedCanBindImmediateValues(G, sfxBlock) },
+    Law("suspended fx can bind suspended values") { suspendedCanBindSuspendedValues(G, sfxBlock) },
+    Law("suspended fx can bind immediate exceptions") { suspendedCanBindImmediateExceptions(pureGen, sfxBlock) },
+    Law("suspended fx can bind suspended exceptions") { suspendedCanBindSuspendedExceptions(pureGen, sfxBlock) }
+  )
+
+  private suspend fun <F, A> nonSuspendedIsLazy(G: Gen<Kind<F, A>>, fxBlock: EagerFxBlock<F, A>) {
+    forAll(G) { f: Kind<F, A> ->
+      val effect = SideEffect()
+      fxBlock {
+        effect.increment()
+        f.bind()
+      }
+
+      effect.counter == 0
+    }
+  }
+
+  private suspend fun <F, A> suspendedIsLazy(G: Gen<Kind<F, A>>, fxBlock: SuspendFxBlock<F, A>) {
+    G.random()
+      .take(1001)
+      .forEach { f ->
+        val effect = SideEffect()
+        fxBlock {
+          effect.increment()
+          f.bind()
+        }
+
+        effect.counter shouldBe 0
+      }
+  }
+
+  private suspend fun <F, A> nonSuspendedCanBindImmediate(G: Gen<Kind<F, A>>, fxBlock: EagerFxBlock<F, A>) {
+    forAll(G) { f: Kind<F, A> ->
+      fxBlock {
+        val res = !f
+        res
+      } == f
+    }
+  }
+
+  private fun <F, A> nonSuspendedCanBindImmediateException(G: Gen<Kind<F, A>>, fxBlock: EagerFxBlock<F, A>) {
+    forAll(G, Gen.throwable()) { f, exception ->
+      shouldThrow<Throwable> {
+        fxBlock {
+          val res = !f
+          throw exception
+          res
+        }
+
+        fail("It should never reach here. fx should've thrown $exception")
+      } == exception
+    }
+  }
+
+  private suspend fun <F, A> suspendedCanBindImmediateValues(G: Gen<Kind<F, A>>, fxBlock: SuspendFxBlock<F, A>) {
+    G.random()
+      .take(1001)
+      .forEach { f ->
+        fxBlock {
+          val res = !f
+          res
+        } shouldBe f
+      }
+  }
+
+  private suspend fun <F, A> suspendedCanBindSuspendedValues(G: Gen<Kind<F, A>>, fxBlock: SuspendFxBlock<F, A>) {
+    G.random()
+      .take(10)
+      .forEach { f ->
+        fxBlock {
+          val res = !(suspend {
+            sleep(100)
```

Comment:
I wrote a set of generic functions `suspended()` that allow you to lift an immediate value, or `Throwable` to a suspended value or exception without having to sleep.

https://github.com/arrow-kt/arrow-fx/blob/master/arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/predef-test.kt#L123